src/Types/ScriptLookups.purs

```diff
@@ -754,19 +755,19 @@ addOwnInput (InputConstraint { txOutRef }) = do
 
 -- | Add a typed output and return its value.
 addOwnOutput
-  :: forall (a :: Type) (b :: Type)
-   . DatumType a b
-  => FromData b
-  => ToData b
-  => OutputConstraint b
-  -> ConstraintsM a (Either MkUnbalancedTxError Unit)
+  :: forall (v :: Type) (r :: Type) (d :: Type)
+   . DatumType v d
+  -- => FromData d
+  => ToData d
+  => OutputConstraint d
+  -> ConstraintsM v (Either MkUnbalancedTxError Unit)
 addOwnOutput (OutputConstraint { datum: d, value }) = do
   networkId <- getNetworkId
   runExceptT do
     ScriptLookups { typedValidator } <- use _lookups
     inst <- liftM TypedValidatorMissing typedValidator
     let value' = fromPlutusValue value
-    typedTxOut <- except $ mkTypedTxOut networkId inst d value'
+    typedTxOut :: TypedTxOut v d <- except $ mkTypedTxOut networkId inst d value'
```

Comment:
Is this annotation needed?

Reply:
Not anymore, it seems. Before there was type ambiguity